include/deal.II/fe/fe_base.h

```diff
@@ -128,6 +128,31 @@ namespace FiniteElementDomination
   inline Domination operator&(const Domination d1, const Domination d2);
 } // namespace FiniteElementDomination
 
+/**
+ * Internal data structure for setting up FiniteElementData.
+ */
+struct PrecomputedFiniteElementData
```

Comment:
Could we give this a more descriptive name, something like `FiniteElementIndexData`?

Reply:
What about generic `GenericDoFsPerObject`?

Reply:
:+1: